components/src/core/drawer/drawer-header/drawer-header.component.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { DrawerHeaderComponent } from './drawer-header.component';
+import { DrawerHeaderModule } from './drawer-header.module';
+import { Component } from '@angular/core';
+import { MatIconModule } from '@angular/material/icon';
+import { By } from '@angular/platform-browser';
+
+@Component({
+	template: `
+        <pxb-drawer-header></pxb-drawer-header>
+    `,
+})
+class TestDrawerHeader { }
+
+@Component({
+	template: `
+        <pxb-drawer-header>
+            <div titleContent id="test-title-content">test title content</div>
+        </pxb-drawer-header>
+    `,
+})
+class TestDrawerHeaderWithTitleContent { }
+
+@Component({
+	template: `
+    		<pxb-drawer-header>
+          <button id="test-icon" mat-icon-button icon>
+            <mat-icon>menu</mat-icon>
+          </button>
+        </pxb-drawer-header>
+    `,
+})
+class TestDrawerHeaderWithIcon { }
+
+describe('DrawerHeaderComponent', () => {
+	let component: DrawerHeaderComponent;
+	let fixture: ComponentFixture<DrawerHeaderComponent>;
+
+	beforeEach(() => {
+		TestBed.configureTestingModule({
+			declarations: [TestDrawerHeader, TestDrawerHeaderWithTitleContent, TestDrawerHeaderWithIcon],
+			imports: [DrawerHeaderModule, MatIconModule],
+		}).compileComponents();
+		fixture = TestBed.createComponent(DrawerHeaderComponent);
+		component = fixture.componentInstance;
+	});
+
+	it('should create', () => {
+		fixture.detectChanges();
+		expect(component).toBeTruthy();
+	});
+
+	it('should render title', () => {
+		component.title = 'test title';
+		fixture.detectChanges();
+		const title = fixture.debugElement.query(By.css('.pxb-drawer-header-title'));
+		expect(title.nativeElement.innerHTML.trim()).toBe('test title');
+	});
+
+	it('should render subtitle', () => {
+		component.subtitle = 'test subtitle';
+		fixture.detectChanges();
+		const subtitle = fixture.debugElement.query(By.css('.pxb-drawer-header-subtitle'));
+		expect(subtitle.nativeElement.innerHTML.trim()).toBe('test subtitle');
+	});
+
+	it('should render titleContent', () => {
+		const customFixture = TestBed.createComponent(TestDrawerHeaderWithTitleContent);
+		customFixture.detectChanges();
+		const content: HTMLElement = customFixture.nativeElement.querySelector('#test-title-content');
+		expect(content.innerHTML).toBe('test title content');
+	});
+
+	it('should render icon', () => {
```

Comment:
Can you also add a test for enforcing class names?  See other components' test for reference.

Reply:
A test has been added to enforce class names.